### PR TITLE
Add support for python 3.11

### LINKED ISSUE #1459: Add support for Python 3.11
### Problem Description
CTGAN should work on Python 3.11

### Background
The SDV library currently runs on Python versions 3.7-3.10. A while ago, we had made the decision to exclude 3.11 because the most recent versions of torch were incompatible with it.

However, torch is now fully upgraded to 2.0 so this should no longer be an issue.

### Dependencies
- CTGAN: https://github.com/sdv-dev/CTGAN/issues/296
- DeepEcho: https://github.com/sdv-dev/DeepEcho/issues/74
- SDMetrics: https://github.com/sdv-dev/SDMetrics/issues/353

### PR DESCRIPTION
**Pull request by fealho**
Resolve #1459.

### CODE REVIEW

**Comment by fealho** [comment]
PR is waiting on CTGAN, SDMetrics and DeepEcho releases with python 3.11.

File: sdv/multi_table/hma.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 import numpy as np
 import pandas as pd
+from packaging import version
```
Comment (amontanez24) [inline sdv/multi_table/hma.py:9]: can we avoid using packaging? It's not listed as an sdv dependency and is probably installed by one of the other libraries. It seems like matplotlib and plotly require it and those are also not direct dependencies for SDV

**Review by amontanez24 — APPROVED** [review]
LGTM!

**Review by frances-h — APPROVED** [review]
Looks good! 👍

**Pull request merged**